configure.py

```diff
@@ -106,10 +106,52 @@ def obtain_ido_recomp(version: str):
     shutil.unpack_archive(target_path, download_dir)
     os.remove(target_path)
 
+def obtain_asm_processor():
+    download_dir = TOOLS_DIR / "asm_proc"
+
+    if download_dir.exists():
+        print(
+            f"asm-processor already exists at {download_dir}, removing and re-downloading"
+        )
+        shutil.rmtree(download_dir)
+
+    ASM_PROC_RECOMP_VERSION = "1.0.1"
+
+    if sys.platform == "darwin":
+        asm_proc_os = "x86_64-apple-darwin"
```

Comment:
does asmproc only support x86?

Reply:
I pushed a change for this. In theory it should download the aarch64 build if the host is running an arm machine, but I don't have any way to test this.